core/services/ocr2/plugins/s4/plugin.go

```diff
@@ -0,0 +1,288 @@
+package s4
+
+import (
+	"context"
+	"errors"
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/services/s4"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+	"google.golang.org/protobuf/proto"
+)
+
+type plugin struct {
+	config          *PluginConfig
+	orm             s4.ORM
+	minQueryAddress *big.Int
+	maxQueryAddress *big.Int
+	addressInteval  *big.Int
+}
+
+type key struct {
+	address string
+	slot    uint
+}
+
+var _ types.ReportingPlugin = (*plugin)(nil)
+
+func NewConsensusPlugin(config *PluginConfig, orm s4.ORM) (*plugin, error) {
```

Comment:
S4 is reflected in the package name, I thought it is redundant to specify s4 in two identifiers.
For user it will look like `s4.NewS4ReportingPlugin`...
`consensus` was a working title, I will follow your recommendation here.